include/javabind/optional.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * javabind: effective C++ and Java interoperability
+ * @see https://github.com/hunyadi/javabind
+ *
+ * Copyright (c) 2024 Levente Hunyadi
+ *
+ * This work is licensed under the terms of the MIT license.
+ * For a copy, see <https://opensource.org/licenses/MIT>.
+ */
+
+#pragma once
+#include "object.hpp"
+#include "signature.hpp"
+#include <optional>
+
+namespace javabind
+{
+    template <typename T>
+    struct ClassTraits<std::optional<T>>
+    {
+        constexpr static std::string_view class_name = arg_type_t<T>::class_name;
+        constexpr static std::string_view class_path = arg_type_t<T>::class_path;
+        constexpr static std::string_view java_name = arg_type_t<T>::java_name;
+    };
+
+    /**
+     * Converts a C++ optional with an object into a Java List.
+     */
+    template <typename T>
+    struct JavaOptionalType : AssignableJavaType<std::optional<T>>
```

Comment:
This is a completely new feature. We might want to update `README.md` to indicate to readers that they can also send/receive `std::optional<T>` just as they can an `std::vector<T>` or an `std::map<K, V>`.

Reply:
Changed to if-else